### PR TITLE
Add some useful navigation for UI

### PR DESCRIPTION
**Pull request by lucas42**
Adds links to items and relationships as primary endpoints.  Other endpoints are in an "Advanced" subsection.

### CODE REVIEW

File: cmdbv2api/views.py [hunk 1]
```diff
@@ -242,7 +242,7 @@ def help(request, **kwargs):
     help['Data Items']['Delete Data Item'] = 'curl -X DELETE ' + baseurl + 'items/<dataTypeid>/<dataItemid>'
 
     help['Relationships'] = {}
-    help['Relationships']['List All Relationships'] = 'curl -X GET ' + baseurl + 'relationships'
+    help['Relationships']['List All Relationships'] = 'curl -X GET ' + baseurl + '/relationships'
```
Comment (GeoffThorpeFT) [inline cmdbv2api/views.py]: slash in wrong place - should be at end of string

File: cmdbv2api/templates/nav.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+	  {% block navigation %}
+      <!-- Navigation -->
+      <div class="o-techdocs-sidebar">
+        <ul class="breadcrumb o-techdocs-nav">
+
+    		<li{% if request.path == "/v2/" %} class="active" aria-selected="true"{% endif %}><a href="/v2/?apikey={{request.user.APIKey}}">Intro</a></li>
+    		<li{% if request.path == "/v2/authentication" %} class="active" aria-selected="true"{% endif %}><a href="/v2/authentication?apikey={{request.user.APIKey}}">API Authentication</a></li>
+    		<li{% if request.path == "/v2/items/" %} class="active" aria-selected="true"{% endif %}><a href="/v2/items/?apikey={{request.user.APIKey}}">Items</a></li>
```
Comment (GeoffThorpeFT) [inline cmdbv2api/templates/nav.html]: This provides navigation to the items endpoint that would display all the items with all their attributes - potentially a huge list. Do we want to encourage the users to do that? We either need a more focused endpoint to present in the navigation or one which allows some form of filtering.

File: cmdbv2api/templates/nav.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    		<li{% if request.path == "/v2/relationships/" %} class="active" aria-selected="true"{% endif %}><a href="/v2/relationships/?apikey={{request.user.APIKey}}">Relationships</a></li>
+    		<li{% if request.path == "/v2/advanced" %} class="active" aria-selected="true"{% endif %}><a href="/v2/advanced?apikey={{request.user.APIKey}}">Advanced Endpoints</a>
+					<ul class="o-techdocs-nav o-techdocs-nav--sub">
+						<li{% if request.path == "/v2/itemtypes/" %} class="active" aria-selected="true"{% endif %}><a href="/v2/itemtypes/?apikey={{request.user.APIKey}}">Item Types</a></li>
+						<li{% if request.path == "/v2/attributetypes/" %} class="active" aria-selected="true"{% endif %}><a href="/v2/attributetypes/?apikey={{request.user.APIKey}}">Attribute Types</a></li>
+						<li{% if request.path == "/v2/relationshiptypes/" %} class="active" aria-selected="true"{% endif %}><a href="/v2/relationshiptypes/?apikey={{request.user.APIKey}}">Relationship Types</a></li>
+						<li{% if request.path == "/v2/itemattributes/" %} class="active" aria-selected="true"{% endif %}><a href="/v2/itemattributes/?apikey={{request.user.APIKey}}">Attributes</a></li>
```
Comment (GeoffThorpeFT) [inline cmdbv2api/templates/nav.html]: This navigation to the itemattributes endpoint will result in a massive output if clicked - every individual attribute of every individual item. A filtered endpoint would be a better option (if we had such a thing!)

File: cmdbv2api/urls.py [hunk 4]
```diff
@@ -26,4 +27,7 @@
     url(r'^v2/itemattributes/(?P<itemTypeID>.+)/(?P<itemID>.+)/?$',                    oneItemAttributes),
     url(r'^v2/cmdbusers/?$',                               listCMDBUsers,           name='cmdbusers'),
     url(r'^v2/cmdbusers/(?P<systemCode>.+)/?$',            CMDBUser,                name='cmdbuserdetail'),
+    url(r'^v2/authentication$',                            docPage,                 {'template':'403'}),
+    url(r'^v2/advanced$',                                  docPage,                 {'template':'advanced'}),
```
Comment (GeoffThorpeFT) [inline cmdbv2api/urls.py]: This is an important concept that we should look to parameterise if possible. An endpoint which references a template with the same name. If we could read these from a table we would have dynamic endpoints driven by the UI needs of our customers.

File: cmdbv2api/views.py [hunk 5]
```diff
@@ -880,3 +880,9 @@ def CMDBUser(request, systemCode):
     if (request.method == 'PUT'):
         request.body = json.loads(request.body.decode(),'utf-8')
     return recordDetail(None, request.method, request.body, CMDBUserSerializer.Meta.model, CMDBUserSerializer, CMDBUserSerializer.Meta.fields, {'systemCode': systemCode}, {'systemCode': systemCode})
+
+# Simple view for handling documentation pages
+from django.views.generic.base import TemplateView
+@api_view() # This decorator makes sure request.user is populated for navigation
+def docPage(request, template='unknowntemplate'):
+    return TemplateView.as_view(template_name=template+'.html')(request)
```
Comment (GeoffThorpeFT) [inline cmdbv2api/views.py]: I believe this is more than a simple view for documentation pages; its a simple view that connects an endpoint with a template of the same name. If we could obtain the endpoint and template from a table then would we be a step towards a flexible user interface with limited need for releases.

File: cmdbv2api/templates/nav.html (hunk 2 above)
Comment (lucas42) [inline cmdbv2api/templates/nav.html]: I think the items endpoint is the most logic place to link to in the navigation.
If we've got issues with the results list being too long, we should address that as a separate issue.  (possibly with filtering or pagination).

File: cmdbv2api/templates/nav.html (hunk 3 above)
Comment (lucas42) [inline cmdbv2api/templates/nav.html]: This is only included in the "advanced" section for completeness, I'm not suggesting anyone actively uses it.  But I think having undocumented endpoints is a bad idea.  If we're not happy with people accessing an endpoint, then we shouldn't have that endpoint.

File: cmdbv2api/urls.py (hunk 4 above)
Comment (lucas42) [inline cmdbv2api/urls.py]: I think I've paramaterised it enough for our current use cases.  I don't want to over-engineer a solution for things we're not doing it yet.  It'll be trivial to refactor this in future.

File: cmdbv2api/views.py (hunk 5 above)
Comment (lucas42) [inline cmdbv2api/views.py]: We can continue this conversation in the email chain.  Don't think it's a requirement for this pull request.

**Pull request merged**